packages/zoe/src/zoeService/types.ts

```diff
@@ -0,0 +1,289 @@
+import type { AnyAmount } from '@agoric/ertp';
+import type { Subscriber } from '@agoric/notifier';
+import type { EReturn } from '@endo/eventual-send';
+import type { ContractStartFunction, StartParams } from './utils.js';
+
+export type IssuerKeywordRecord = Record<Keyword, Issuer<any>>;
+
+type IsAny<T> = 0 extends 1 & NoInfer<T> ? true : false;
+
+/**
+ * Zoe provides a framework for deploying and working with smart
+ * contracts. It is accessed as a long-lived and well-trusted service
+ * that enforces offer safety for the contracts that use it. Zoe has a
+ * single `invitationIssuer` for the entirety of its lifetime. By
+ * having a reference to Zoe, a user can get the `invitationIssuer`
+ * and thus validate any `invitation` they receive from someone else.
+ *
+ * Zoe has two different facets: the public Zoe service and the
+ * contract facet (ZCF). Each contract instance has a copy of ZCF
+ * within its vat. The contract and ZCF never have direct access to
+ * the users' payments or the Zoe purses.
+ */
+export type ZoeService = {
```

Comment:
I'm just skimming the conversion of these type declarations from JSDoc to .ts.

It seems unlikely that errors would be introduced without being caught by tsc.

Reply:
I used a build step to make them, then tweaked